sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1191,6 +1191,12 @@ class Dataset[T] private[sql](
   def col(colName: String): Column = colName match {
     case "*" =>
       Column(ResolvedStar(queryExecution.analyzed.output))
+    case ParserUtils.escapedIdentifier(columnNameRegex)
```

Comment:
@janewangfb Based on the above comment from @cloud-fan and @hvanhovell , how about creating a new function `colRegex`? 

`spark.sql.parser.quotedRegexColumnNames` can be defined as a SQL Parser specific configuration. That configuration will not affect the new function `colRegex`.